src/main/java/wooteco/subway/line/service/LineService.java

```diff
@@ -1,81 +1,79 @@
 
-    private static final Logger log = LoggerFactory.getLogger(LineService.class);
-
-    private final LineRepository lineRepository;
-
-    public LineService(LineRepository lineRepository) {
-        this.lineRepository = lineRepository;
+    public LineService(SectionService sectionService, LineDao lineDao) {
+        this.sectionService = sectionService;
+        this.lineDao = lineDao;
     }
 
+    @Transactional
     public LineResponse save(LineRequest lineRequest) {
         validateLineName(lineRequest);
-        Line line = lineRequest.toEntity();
-        Line newLine = lineRepository.save(line);
-        log.info("{} 노선 생성 성공", newLine.getName());
+        Line newLine = lineDao.save(lineRequest.toEntity());
+        sectionService.save(newLine, lineRequest);
         return new LineResponse(newLine);
     }
 
-    private void validateLineName(LineRequest lineRequest) {
-        if (checkNameDuplicate(lineRequest)) {
+    private void validateLineName(LineRequest lineReq) {
+        if (checkNameDuplicate(lineReq)) {
             throw new DuplicatedNameException("중복된 이름의 노선이 존재합니다.");
         }
+        sectionService.validateExistStations(lineReq.getUpStationId(), lineReq.getDownStationId());
     }
 
     private boolean checkNameDuplicate(LineRequest lineRequest) {
-        return lineRepository.validateDuplicateName(lineRequest.getName());
+        return lineDao.isExistingName(lineRequest.getName());
     }
 
+    @Transactional(readOnly = true)
     public LineResponse findById(Long id) {
-        Line newLine = lineRepository.findById(id);
-        log.info("{} 노선 조회 성공", newLine.getName());
-        return new LineResponse(newLine);
+        Line line = lineDao.findById(id);
+        List<Long> stationIds = sectionService.findAllSectionsId(id);
+        List<StationResponse> stations = sectionService.findStationsByIds(stationIds);
+        return new LineResponse(line, stations);
     }
 
+    @Transactional(readOnly = true)
     public List<LineResponse> findAll() {
-        List<Line> lines = lineRepository.findAll();
-        log.info("지하철 모든 노선 조회 성공");
+        List<Line> lines = lineDao.findAll();
         return lines.stream()
                 .map(LineResponse::new)
                 .collect(Collectors.toList());
     }
 
+    @Transactional
     public void update(Long id, LineRequest lineRequest) {
-        Line updatedLine = validatesRequest(id, lineRequest);
-        lineRepository.update(updatedLine);
-        log.info("노선 정보 수정 완료");
-    }
-
-    private Line validatesRequest(Long id, LineRequest lineRequest) {
-        Line currentLine = lineRepository.findById(id);
-
-        validateUsableName(lineRequest.getName(), currentLine.getName());
-        return new Line(id, lineRequest.getName(), lineRequest.getColor());
+        Line currentLine = lineDao.findById(id);
+        validatesChangeName(lineRequest.getName(), currentLine.getName());
+        currentLine.update(lineRequest);
```

Comment:
```suggestion
        currentLine.update(lineRequest.getName(), currentLine.getName());
```
LineRequest는 dto 입니다. dto와 도메인 객체 사이에 의존도가 생기면 안됩니다.
dto가 새로 생길 때마다 도메인 객체도 변경사항이 생길 수 있습니다.